File: src-frontend/src/types/task.ts
import type { Message } from "./message";

export type TaskType = "agent" | "orchestration";

export type TaskUsage = {
  input_tokens: number;
  output_tokens: number;
  total_tokens: number;
  max_tokens: number;
};

// --- --- --- --- --- ---

export type TaskBase = {
  title: string;
  type: TaskType;
  messages: Message[];
};

export type TaskRead = TaskBase & {
  id: number;
  last_run_at: number;
  agent_id: number | null;
  workspace_id: number;
};

export type TaskCreate = TaskBase & {
  agent_id: number;
  workspace_id: number;
};

export type TaskUpdate = Partial<TaskCreate>;

File: src-server/src/db/schemas/task.py
from . import DTOBase
from ..models.task import TaskMessage, TaskType

class TaskBase(DTOBase):
    title: str
    type: TaskType
    messages: list[TaskMessage]

class TaskRead(TaskBase):
    id: int
    last_run_at: int
    agent_id: int | None = None
    workspace_id: int

class TaskCreate(TaskBase):
    agent_id: int
    workspace_id: int

class TaskUpdate(DTOBase):
    title: str | None = None
    last_run_at: int
    agent_id: int | None = None
    messages: list[TaskMessage] | None = None
